msnc/encoder.py

```diff
@@ -82,3 +106,43 @@ def _view(self, H):
     def _unsort(self, H, indices):
         _, unsorted_indices = torch.tensor(indices).sort()
         return H.index_select(0, unsorted_indices)
+
+
+class AverageEncoder(BaseEncoder):
+
+    def __init__(
+        self,
+        xdim,
+        edim,
+        dropout=0.2,
+        **kwargs
+    ):
+        """Averaging word vectors encoder
+
+        Arguments:
+            xdim {int} -- input feature dimension
+            edim {int} -- embedding dimension
+
+        Keyword Arguments:
+            dropout {float} -- dropout ratio (default: {0.2})
+        """
+        super(AverageEncoder, self).__init__()
+        self.util = Util()
+        self.pad_index = self.util.PAD_INDEX
+        self.xdim = xdim
+        self.edim = edim
+        self.dropout = dropout
+        self.use_cuda = self.util.use_cuda
+        self.embedding = self._init_embedding()
+
+    def forward(self, X):
+        H = []
+        for x in X:
+            h = self._embed(self.util.tensorize(x))
+            h = h.mean(dim=0)
+            H.append(h)
+
+        return torch.stack(H, dim=0)
+
+    def _embed(self, X):
+        return self.embedding(X)
```

Comment:
We can merge the method with `_embed` of `RecurrentEncoder`.

Reply:
Ah, you're right, thanks! :bow: